toolchain/check/import_ref.cpp

```diff
@@ -1568,6 +1570,131 @@ class ImportRefResolver {
         context_.types().GetConstantId(interface_val.type_id()));
   }
 
+  // Forms a constant ID that represents the given instruction ID that denotes
+  // an impl declaration. Impl declarations don't have a constant value, but for
+  // the purposes of importing we pretend that they do.
+  //
+  // TODO: Consider alternative options here. We could make impl declarations
+  // actually evaluate to themselves to avoid this hack.
```

Comment:
I'm fine with this as a TODO, but I would've thought impl decls should be constant. Maybe it hasn't come up just because we mostly talk about them in terms of the constraints they satisfy? I wonder if metaprogramming would end up forcing this.